jobs/build.Jenkinsfile

```diff
@@ -265,9 +265,10 @@ lock(resource: "build-${params.STREAM}") {
             // Nothing changed since the latest build. Check if it's missing
             // some arches and retrigger `build-arch` only for the missing
             // arches, and the follow-up `release` job. Match the exact src
-            // config commit that was used. Skip if not uploading since it's
-            // required for multi-arch.
-            if (uploading) {
+            // config commit that was used. But only do this if there isn't
+            // already a release job pending for that build ID. Skip if not
+            // uploading since it's required for multi-arch.
+            if (uploading && !is_locked("release-${buildID}")) {
```

Comment:
I think this doesn't really give us what you want because of 0664cd6. i.e. the release job could be not running yet because we're waiting on the mArch jobs. I guess we could check all of those locks here.

Reply:
Good catch!